crc_errors/CRC_Count_check.py

```diff
@@ -7,117 +7,130 @@
-                dev.name)))
-            abstract = Lookup.from_device(dev)
-            intf = abstract.ops.interface.interface.Interface(dev)
-            intf.learn()
-            self.all_interfaces[dev.name] = intf.info
-
-    # Second test section
-    @ aetest.test
-    def check_CRC(self):
 
+        list_of_interfaces=self.interface_info.info.keys()
         mega_dict = {}
+        mega_dict[dev.name] = {}
         mega_tabular = []
-        for device, ints in self.all_interfaces.items():
-            mega_dict[device] = {}
-            for name, props in ints.items():
-                counters = props.get('counters')
-                if counters:
-                    smaller_tabular = []
-                    if 'in_crc_errors' in counters:
-                        mega_dict[device][name] = counters['in_crc_errors']
-                        smaller_tabular.append(device)
-                        smaller_tabular.append(name)
-                        smaller_tabular.append(str(counters['in_crc_errors']))
-                        if counters['in_crc_errors']:
-                            smaller_tabular.append('Failed')
-                        else:
-                            smaller_tabular.append('Passed')
+        passing=0
+
+
+        for ints, props in self.interface_info.info.items():
+            counters = props.get('counters')
+            if counters:
+                smaller_tabular = []
+                if 'in_crc_errors' in counters:
+                    mega_dict[dev.name][ints] = counters['in_crc_errors']
+                    smaller_tabular.append(dev.name)
+                    smaller_tabular.append(ints)
+                    smaller_tabular.append(str(counters['in_crc_errors']))
+                    if counters['in_crc_errors']:
+                        smaller_tabular.append('Failed')
+                        passing=1
                     else:
-                        mega_dict[device][name] = None
-                        smaller_tabular.append(device)
-                        smaller_tabular.append(name)
-                        smaller_tabular.append('N/A')
-                        smaller_tabular.append('N/A')
-                mega_tabular.append(smaller_tabular)
-
-        mega_tabular.append(['-'*sum(len(i) for i in smaller_tabular)])
-
-        log.info(tabulate(mega_tabular,
+                        smaller_tabular.append('Passed')
+
+                else:
+                    mega_dict[dev.name][ints] = None
+                    smaller_tabular.append(dev.name)
+                    smaller_tabular.append(ints)
+                    smaller_tabular.append('N/A')
+                    smaller_tabular.append('N/A')
+            mega_tabular.append(smaller_tabular)
+        mega_tabular.append(['-' * sum(len(i) for i in smaller_tabular)])
+
+        #pass megadict to interface test function
+        self.parent.parameters.update(mega=mega_dict[dev.name])
```

Comment:
since this mega dict is just used for this testcase, you can just store it as an attribute under `self`
the testcase class instance `self` persists through this testcase run (this iteration of this class)

Reply:
Gotcha. Brilliant.  So no need to pass attributes up to parent, just declare them as 'self' in the Class.